R/classification.R

```diff
@@ -33,13 +33,12 @@ classification <- function(lcl, ucl, threshold, reference = 0) {
               is.numeric(threshold), noNA(threshold), is.number(reference),
               noNA(reference), all(lcl <= ucl))
   if (length(threshold) == 1) {
-    assert_that(-abs(threshold) < reference, reference < abs(threshold))
```

Comment:
please fix this assertion. It should be `assert_that(threshold > 0)`

Reply:
Is this needed? In line 33 we have an assertion that checks if threshold is numeric and in line 36 we have a multiplication by `abs(threshold)` and thus giving one negative threshold value to the function argument will be treated correctly (you catched that use-case already). Also, from a user perspective, it can happen easily that you pass a negative threshold if you are thinking on another scale than the one that needs to be passed to the threshold argument (e.g. `threshold = log(3/4)`).

Reply:
I see your point. But shouldn't we test that the threshold is at least different from zero? Otherwise the threshold become identical to the reference.

Reply:
IMO, this is not necessary. There may be cases where you explicitly don't want to use thresholds and that can be achieved by setting `threshold = 0`. The classification than simplifies to `++, ?, --`.  If we test for threshold different from zero, and someone has such a use-case, ugly workarounds will be used such as `threshold = 10^-20`. Therefore, I think it is best to just allow this.